### PR TITLE
Mir borrowck as query

### PR DESCRIPTION
**Pull request by pnkfelix**
Turn the `mir-borrowck` pass (aka "transform") into a query.

(If I had realized how relatively easy this was going to be, I would have made it part of #43108. `let hindsight = 20/20;`)

### CODE REVIEW

File: src/librustc/ty/maps.rs [hunk 1]
```diff
@@ -923,6 +923,8 @@ define_maps! { <'tcx>
     [] coherent_trait: coherent_trait_dep_node((CrateNum, DefId)) -> (),
 
     [] borrowck: BorrowCheck(DefId) -> (),
+    // FIXME: shouldn't this return a `Result<(), BorrowckErrors>` instead?
```
Comment (arielb1) [inline src/librustc/ty/maps.rs]: We'll figure out the borrowck error story.

**Comment by kennytm** [comment]
@bors retry 

Build canceled without any reason.

**Pull request merged**